src/openforms/contrib/kadaster/api/tests/test_api_map_search.py

```diff
@@ -0,0 +1,345 @@
+            return_value=KadasterApiConfig(kadaster_service=self.kadaster_service),
+        )
+        self.config_mock = config_patcher.start()
+        self.addCleanup(config_patcher.stop)
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.submission = SubmissionFactory.create()
+
+        cls.kadaster_service = ServiceFactory(
+            api_root="https://kadaster/",
+            oas="https://kadaster/api/schema/openapi.yaml",
+        )
+
+    @requests_mock.Mocker()
+    def test_call_with_query_parameter_utrecht(self, m):
+        mock_service_oas_get(
+            m,
+            url="https://kadaster/",
+            service="locatieserver_openapi",
+            oas_url="https://kadaster/api/schema/openapi.yaml",
+        )
+        m.get(
+            "https://kadaster/v3_1/free",
+            status_code=200,
+            json={
+                "response": {
+                    "numFound": 1,
+                    "start": 0,
+                    "maxScore": 9.7794895,
+                    "numFoundExact": True,
+                    "docs": [
+                        {
+                            "bron": "Bestuurlijke Grenzen",
+                            "identificatie": "0344",
+                            "provinciecode": "PV26",
+                            "type": "gemeente",
+                            "provincienaam": "Utrecht",
+                            "centroide_ll": "POINT(5.0747543 52.09113798)",
+                            "gemeentecode": "0344",
+                            "weergavenaam": "Gemeente Utrecht",
+                            "provincieafkorting": "UT",
+                            "centroide_rd": "POINT(133587.182 455921.594)",
+                            "id": "gem-df0ca8ab37eccea5217e2a13f74d2833",
+                            "gemeentenaam": "Utrecht",
+                            "score": 9.7794895,
+                        },
+                    ],
+                }
+            },
+        )
+
+        url = reverse("api:geo:address-search")
+        self._add_submission_to_session(self.submission)
+
+        response = self.client.get(url, {"q": "utrecht"})
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        body = response.json()
+
+        self.assertEqual(
+            body,
+            [
+                {
+                    "label": "Gemeente Utrecht",
+                    "latLng": {"lat": 52.09113798, "lng": 5.0747543},
+                    "rd": {"x": 133587.182, "y": 455921.594},
+                },
+            ],
+        )
+
+    def test_call_with_no_active_submission(self):
+        url = reverse("api:geo:address-search")
+
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @patch(
+        "openforms.contrib.kadaster.api.views.KadasterApiConfig.get_solo",
+        return_value=KadasterApiConfig(kadaster_service=None),
+    )
```

Comment:
I tried doing this but it didn't work. Thats why I ended up overwriting the patch.

I ended up deleting the test and related code because the model field is required and has a default value so the code would never be hit.